content/spotworkers/workers.md

```diff
@@ -0,0 +1,88 @@
+nodes that will connect to the EKS cluster.
+
+This template will create a single ASG that leverages the latest feature to mix multiple instance types and purchase as a single K8s nodegroup. Check out this blog: [New – EC2 Auto Scaling Groups With Multiple Instance Types & Purchase Options](https://aws.amazon.com/tw/blogs/aws/new-ec2-auto-scaling-groups-with-multiple-instance-types-purchase-options/) for details.
+
+#### Retrieve the Worker Role name
+
+First, we will need to collect the Role Name that is in use with our EKS worker nodes
+
+```bash
+INSTANCE_PROFILE_PREFIX=$(aws cloudformation describe-stacks --stack-name eksctl-eksworkshop-eksctl-nodegroup-0 | jq -r '.Stacks[].Outputs[].ExportName' | sed 's/:.*//')
+INSTANCE_PROFILE_NAME=$(aws iam list-instance-profiles | jq -r '.InstanceProfiles[].InstanceProfileName' | grep $INSTANCE_PROFILE_PREFIX)
+ROLE_NAME=$(aws iam get-instance-profile --instance-profile-name $INSTANCE_PROFILE_NAME | jq -r '.InstanceProfile.Roles[] | .RoleName')
+echo $ROLE_NAME
+```
+
+Copy the Role Name for use as a Parameter in the next step
+
+```text
+# Example Output
+eksctl-eksworkshop-eksctl-nodegro-NodeInstanceRole-XXXXXXXX
+```
+
+#### Launch the CloudFormation Stack
+We will launch the CloudFormation template as a new set of worker nodes, but it's also possible to update the nodegroup CloudFormation stack created by the *eksctl* tool.
+
+Click the **Launch** button to create the CloudFormation stack in the AWS Management Console.
+
+| Launch template |  |  |
+| ------ |:------:|:--------:|
+| EKS Workers - Spot and On Demand |  {{% cf-launch "amazon-eks-nodegroup-with-mixed-instances.yml?stackName=eksworkshop-nodegroup-0" %}} | {{% cf-download "amazon-eks-nodegroup-with-mixed-instances.yml" %}}  |
+
+{{% notice tip %}}
+Confirm the region is correct based on where you've deployed your cluster.
+{{% /notice %}}
+Once the console is open you will need to configure the missing parameters. Use the table below for guidance.
+
+| Parameter | Value |
+|-----------|-------|
+| Stack Name: | eksworkshop-spot-workers |
+| Cluster Name: | eksworkshop-eksctl (or whatever you named your cluster) |
+|ClusterControlPlaneSecurityGroup: | Select from the dropdown. It will contain your cluster name and the words **'ControlPlaneSecurityGroup'** |
+|NodeInstanceRole: | Use the same role name that is used with to the other EKS worker nodes. (e.g. eksctl-eksworkshop-eksctl-nodegro-NodeInstanceRole-XXXXX)
+|NodeImageId: | Visit this [**link**](https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html) and select the non-GPU image for your region - **Check for empty spaces in copy/paste**|
+|KeyName: | SSH Key Pair created earlier or any valid key will work |
+|VpcId: | Select your workshop VPC from the dropdown |
+|Subnets: | Select the **public** subnets for your workshop VPC from the dropdown |
+|OnDemandBootstrapArguments: | --kubelet-extra-args --node-labels=lifecycle=OnDemand |
```

Comment:
On cfn console it is BootstrapArgumentsForOnDemand instead of OnDemandBootstrapArguments
and BootstrapArgumentsForSpotFleet instead of SpotBootstrapArguments. Either should be updated to sync.